internal/clients/nightfall/nightfall_test.go

```diff
@@ -1,347 +1,163 @@
-				createComment(creditCard2Response),
-			},
-			desc: "credit cards omit api finding",
-		},
-		{
-			haveContentToScanList: []*contentToScan{
-				createContentToScan(creditCardNumberContent),
-				createContentToScan("nothing in here"),
-				createContentToScan(apiKeyContent),
-			},
-			haveScanResponseList: [][]nightfallAPI.ScanResponse{
-				{
-					creditCardResponse,
-				},
-				{
-					createScanResponse("low likelihood on 4534343", nightfallAPI.CREDIT_CARD_NUMBER, nightfallAPI.UNLIKELY),
-				},
-			},
-			haveTokenExclusionSet: emptyTokenExclusionSet,
-			want: []*diffreviewer.Comment{
-				createComment(creditCardResponse),
-			},
-			desc: "single credit card passing likelihood threshold",
-		},
+	scanResp := [][]nightfallAPI.ScanResponse{
+		{},
 		{
-			haveContentToScanList: []*contentToScan{
-				createContentToScan("nothing in here"),
-				createContentToScan("nothing in here"),
-				createContentToScan("nothing in here"),
-				createContentToScan("nothing in here"),
-				createContentToScan(apiKeyContent),
-			},
-			haveScanResponseList: [][]nightfallAPI.ScanResponse{
-				{},
-				{},
-				{},
-				{},
-				{
-					apiKeyResponse,
+			{
+				Fragment: exampleCreditCardNumber,
+				Detector: string(nightfallAPI.CREDIT_CARD_NUMBER),
+				Confidence: nightfallAPI.ScanResponseConfidence{
+					Bucket: string(nightfallAPI.LIKELY),
 				},
 			},
-			haveTokenExclusionSet: emptyTokenExclusionSet,
-			want:                  []*diffreviewer.Comment{},
-			desc:                  "no comments",
-		},
-		{
-			haveContentToScanList: []*contentToScan{
-				createContentToScan(creditCardNumberContent),
-				createContentToScan("nothing in here"),
-				createContentToScan(apiKeyContent),
-				createContentToScan(creditCardNumber2Content),
-			},
-			haveScanResponseList: [][]nightfallAPI.ScanResponse{
-				{
-					creditCardResponse,
-				},
-				{},
-				{
-					apiKeyResponse,
-				},
-				{
-					creditCard2Response,
-				},
-			},
-			haveTokenExclusionSet: tokenExclusionSet,
-			want: []*diffreviewer.Comment{
-				createComment(creditCardResponse),
-			},
-			desc: "single credit card excluded",
 		},
+		{},
+	}
+	c := diffreviewer.Comment{
+		FilePath:   filePath,
+		LineNumber: lineNum,
+		Body:       fmt.Sprintf("Suspicious content detected (%s, type %s)", blurredCreditCard, nightfallAPI.CREDIT_CARD_NUMBER),
+		Title:      fmt.Sprintf("Detected %s", nightfallAPI.CREDIT_CARD_NUMBER),
 	}
+	expectedComments := []*diffreviewer.Comment{&c, &c, &c}
 
-	for _, tt := range tests {
-		actual := createCommentsFromScanResp(tt.haveContentToScanList, tt.haveScanResponseList, detectorConfigs, tt.haveTokenExclusionSet)
-		assert.Equal(t, tt.want, actual, fmt.Sprintf("Incorrect response from createCommentsFromScanResp: %s test", tt.desc))
+	for i := 0; i < numScanReq; i++ {
+		mockAPIClient.EXPECT().ScanAPI().Return(mockScanAPI)
+		mockScanAPI.EXPECT().ScanPayload(gomock.Any(), gomock.Any()).Return(scanResp, nil, nil)
```

Comment:
any way we can do stricter type checking for the 2nd val passed to ScanPayload here?